### PR TITLE
Bug 1821437 - Upgrade to Xcode 14.0

### PR DESCRIPTION
**Pull request by badboy**
This is essentially a revert of the revert-commit f941d76 again.

---

We can hold off until our consumers are good to go with this.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 * Kotlin
   * Increase to Android target/compile SDK version 33 ([#2246](https://github.com/mozilla/glean/pull/2246))
+* iOS
+  * Glean for iOS is now being built with Xcode 14.0 ([#](https://github.com/mozilla/glean/pull/TODO))
```
Comment (Dexterp37) [inline CHANGELOG.md:8]: Please update this:)

**Review by Dexterp37 — APPROVED** [review]
r+wc

**Comment by rvandermeulen** [comment]
@badboy IIUC, Firefox and Focus iOS are both on XCode 14.3 now. Are there other consumers that need to update or can this land now? I guess AppServices is still on 13.4.1, though. Not sure if that blocks or not.

File: .circleci/config.yml [hunk 2]
```diff
@@ -513,14 +513,14 @@ jobs:
       - setup-rust-toolchain
       - restore_cache:
           name: Restore rubygems cache
-          key: swift-docs-gems-v13
+          key: swift-docs-gems-v15
```
Comment (badboy) [inline .circleci/config.yml:516]: We used `-v14` in the reverted commit before. Thus skipping that and going for v15 right away.

**Review by rosahbruno — APPROVED** [review]

**Pull request merged**